src/Movies/store.js

```diff
@@ -0,0 +1,32 @@
+import React, { useContext, useReducer, useMemo, createContext } from "react";
+import { movieReducer } from "./reducer";
+
+const MovieContext = createContext();
+
+const defaultState = {
+  movies: [],
+  selectedMovie: null,
+  movieLoading: false,
+  movieError: ""
+};
+
+const MovieProvider = props => {
+  const [state, dispatch] = useReducer(movieReducer, defaultState);
+  const value = useMemo(() => [state, dispatch], [state]);
```

Comment:
this is (useMemo) overreliance on react antipatterns and premature optimization, please don't fall victim to this trap. it is one thing learning react's new apis and toolbelts, but another thing quickly jumping on the gun with new antipatterns emerging.